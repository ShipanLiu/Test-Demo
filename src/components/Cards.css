.cards {
  /* padding: 4rem; */
  background: #fff;
}

.cards__container {
  display: flex;
  align-items: center;
  width: 100%;
  /* max-width: 1120px; */
  margin: 0 auto;
}

.cards__wrapper {
  position: relative;
  margin: 0.5rem 0 1rem;
  border: 1px solid #000;
}

.cards__items {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 24px;
}

.cards__item {
  flex: 1 0 21%;
  margin: 0 1rem;
  border: 1px solid #000;
  border-radius: 10px;
}

.cards__item__link {
  display: flex;
  flex-flow: column;
  width: 100%;
  overflow: hidden;
  text-decoration: none;
  border-radius: 10px;
  box-shadow: 0 6px 20px rgba(56, 125, 255, 0.17);
  -webkit-filter: drop-shadow(0 6px 20px rgba(56, 125, 255, 0.017));
  filter: drop-shadow(0 6px 20px rgba(56, 125, 255, 0.017));
}

.cards__item__pic-wrap {
  position: relative;
  width: 100%;
  padding-top: 100%;
  overflow: hidden;
}

.fade-img {
  animation-name: fade-img;
  animation-duration: 2s;
}
.sale-lable {
  position: absolute;
  top: 0;
  box-sizing: border-box;
  max-width: calc((100%) - 60px);
  margin-left: 10px;
  padding: 6px 8px;
  color: #fff;
  font-weight: 700;
  font-size: 12px;
  background-color: #1f98f4;
  content: attr(data-category);
}

.cards__item__img {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: block;
  width: 100%;
  max-width: 100%;
  height: 100%;
  max-height: 100%;
  object-fit: contain;
  transition: all 0.2s linear;
}

.cards__item__img:hover {
  transform: scale(1.1);
}

.cards__item__info {
  padding: 20px 30px 30px;
}

.cards__item__text {
  color: #252e48;
  font-size: 18px;
  line-height: 24px;
}

@media only screen and (min-width: 1200px) {
  .content__blog__container {
    width: 84%;
  }
}

@media only screen and (min-width: 1024px) {
  .cards__items {
    display: flex;
  }
}

@media only screen and (max-width: 1024px) {
  .cards__item {
    margin-bottom: 2rem;
  }
}
