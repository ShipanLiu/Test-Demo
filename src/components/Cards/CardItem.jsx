import React from 'react'
import { Link } from 'react-router-dom'

function CardItem(props) {
  const { price, src, name, path } = props
  return (
    <>
      <li className="cards__item">
        <Link className="cards__item__link" to={path}>
          <div className="cards__item__pic-wrap">
            <img className="cards__item__img" alt="img not found" src={src} />
            {props.isSale ? <span className="sale-lable">Sale</span> : ''}
          </div>
          <div className="cards__item__info">
            <h5 className="cards__item__text">{name}</h5>
            <h5 className="cards__item__text">{price}</h5>
          </div>
        </Link>
      </li>
    </>
  )
}

export default CardItem
