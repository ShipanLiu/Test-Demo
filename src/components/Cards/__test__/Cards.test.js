import React from 'react'
import ReactDOM from 'react-dom'
import Cards from '../Cards'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Cards></Cards>, div)
})

// test wheather the 'onSearch' property was delivered
it('renders perfectly', () => {
  const { getByTestId } = render(<Cards choosedItem></Cards>)
  expect(getByTestId('cards')).toHaveProperty
})

it('matches snapshot', () => {
  const tree = renderer.create(<Cards choosedItem></Cards>).toJSON()
  console.log(tree)
  // create a Snapshot file
  expect(tree).toMatchSnapshot()
})
