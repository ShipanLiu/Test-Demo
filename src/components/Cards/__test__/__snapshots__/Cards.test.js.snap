// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="cards"
  data-testid="cards"
>
  <div
    className="cards__container"
  >
    <div
      className="cards__wrapper"
    >
      <ul
        className="cards__items"
      />
    </div>
  </div>
</div>
`;
